src/Apps/Order2/Routes/Details/Components/Order2DetailsFulfillmentInfo.tsx

```diff
@@ -16,10 +16,23 @@ export const Order2DetailsFulfillmentInfo: React.FC<
   const { fulfillmentDetails, selectedFulfillmentOption, shippingOrigin } =
     orderData
 
-  if (!selectedFulfillmentOption) return null
-
+  const isBlankFulfillmentDetails =
+    !fulfillmentDetails ||
+    Object.values(fulfillmentDetails).every(
+      part =>
+        !part ||
+        part === undefined ||
+        (typeof part === "string" && part.trim() === ""),
+    )
   const isPickup = selectedFulfillmentOption?.type === "PICKUP"
 
+  if (
+    (isPickup && !shippingOrigin) ||
+    (!isPickup && isBlankFulfillmentDetails)
+  ) {
+    return null
+  }
```

Comment:
Only other comment here is that maybe we always want to show something, even if it is incomplete. Like would just the word 'pickup' be preferable if we don't have a pickup location/shipping origin?

Reply:
That is a good question. I thought about that too. 'Pickup' does make sense a bit but the 'Ship to' does not without some shipping data in my mind. But maybe we can do more generic name like 'Shipping' or 'Ship'. Will bring it up to confirm at checkup today.